### PR TITLE
Improve reinstalling of the hosts

### PR DESCRIPTION
**Pull request by machacekondra**
This patch improve the reinstalling of the host:

 - support to pass also public key instead of password
 - improve the search, so we search only for host in install_failed state and
   only hosts we specified in variables
 - document the timeouts

### CODE REVIEW

**Comment by machacekondra** [comment]
@mwperina

File: roles/ovirt-hosts/README.md [hunk 1]
```diff
@@ -14,15 +14,16 @@ Role Variables
...
-| cluster       | UNDEF          | The cluster that the host must connect to.    |
-| timeout       | 1200           | Maximum wait time for the host to be in an UP state.  |
-| poll_interval | 20             | Polling interval to check the host status. |
+| Name          | Default value    | Description                           |
+|---------------|------------------|---------------------------------------|
+| name          | UNDEF (Required) | Name of the host.                      |
+| state         | present          | Specifies whether the host is `present` or `absent`.  |
+| address       | UNDEF (Required) | IP address or FQDN of the host.   |
+| password      | UNDEF            | The host's root password. Required if <i>public_key</i> is false. |
+| public_key    | UNDEF            | If <i>true</i> the public key should be used to authenticate to host. |
```
Comment (mwperina) [inline roles/ovirt-hosts/README.md:23]: I thought that this param will contain the public key of the host. So why boolean?

Comment (machacekondra) [reply roles/ovirt-hosts/README.md:23]: No, it's true/false, we don't support specify custom public key, only the default one - the one deployed by host-deploy proccess.

Comment (mwperina) [reply roles/ovirt-hosts/README.md:23]: Ahh, right

**Review by mwperina — APPROVED** [review]
👍

**Pull request merged**